promotion/datastore.go

```diff
@@ -680,6 +684,93 @@ func (pg *Postgres) SaveClaimCreds(creds *ClaimCreds) error {
 	return err
 }
 
+// GetCustodianDrainInfo Get the status of the custodian drain info
+func (pg *Postgres) GetCustodianDrainInfo(paymentID *uuid.UUID) ([]CustodianDrain, error) {
+	resp := []CustodianDrain{}
+	// get the linked wallet info
+	stmt := `
+select
+	user_deposit_account_provider, user_deposit_destination
+from
+	wallets
+where
+	id = $1
+`
+	var custodian Custodian
+	if err := pg.RawDB().Get(&custodian, stmt, paymentID); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	// get all of the drain jobs for this payment id
+	stmt = `
+select
+	batch_id,
+	split_part(credentials->0->>'issuer',':',1) as promotion_id,
+	completed_at,
+	json_array_length(credentials)*0.25 as value,
+	case when erred then 'errored' else 'succeeded' end as state,
+	errcode,
+	transaction_id
+from
+	claim_drain
+where
+	wallet_id = $1
```

Comment:
I don't think this is indexed (I'm just grepping through the migrations so I could be wrong, wish we had a schema file)

Reply:
good catch.  resolved: https://github.com/brave-intl/bat-go/pull/820/commits/4160c96c3f6f88ec414f630e96ffb51bb9310455